Sources/Sentry/SentryTracer.m

```diff
@@ -193,7 +193,7 @@ - (instancetype)initWithTransactionContext:(SentryTransactionContext *)transacti
     if (_configuration.profilesSamplerDecision.decision == kSentrySampleDecisionYes
         || sentry_isTracingAppLaunch) {
         _internalID = [[SentryId alloc] init];
-        if ((_isProfiling = [SentryProfiler startWithTracer:_internalID])) {
+        if ((_isProfiling = [SentryLegacyProfiler startWithTracer:_internalID])) {
```

Comment:
I really like this refactoring approach. Almost no changes in the SentryTracer so far, which minimizes the risk of bugs for the existing implementation of the SentryTracer 💯